src/Areas/AzureArc/Commands/DisconnectFromAzureArcCommand.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using AzureMcp.Areas.AzureArc.Services;
+using AzureMcp.Options.Arc;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.Commands.Arc;
+
+public sealed class DisconnectFromAzureArcCommand(ILogger<DisconnectFromAzureArcCommand> logger) : GlobalCommand<ArcConnectOptions>
+{
+    private const string _commandTitle = "Disconnects a cluster from Azure Arc";
+
+    // Define options as fields
+    private readonly Option<string> _clusterNameOption = new("--cluster-name", "Name of the cluster to disconnect from Azure Arc");
+    private readonly Option<string> _resourceGroupNameOption = new("--resource-group-name", "Name of the resource group");
+    private readonly Option<string> _userProvidedPathOption = new("--user-provided-path", "Path to the user-provided files");
+
+    public override string Name => "disconnect-from-azure-arc";
+
+    public override string Description =>
+        "Disconnects an arc connected cluster from Azure Arc. It deletes all extensions and removes the cluster from Arc. The --user-provided-path option is required.";
+
+    public override string Title => _commandTitle;
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+
+        // Add options to the command
+        command.AddOption(_clusterNameOption);
+        command.AddOption(_resourceGroupNameOption);
+        command.AddOption(_userProvidedPathOption);
+    }
+
+    protected override ArcConnectOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+
+        // Use the defined Option instances
+        options.ClusterName = parseResult.GetValueForOption(_clusterNameOption);
+        options.ResourceGroupName = parseResult.GetValueForOption(_resourceGroupNameOption);
+        options.UserProvidedPath = parseResult.GetValueForOption(_userProvidedPathOption);
+
+        return options;
+    }
+
+    [McpServerTool(
+        Destructive = true,
+        ReadOnly = false,
+        Title = _commandTitle)]
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        try
+        {
+            // Retrieve options using model binding
+            var options = BindOptions(parseResult);
+
+            var arcService = context.GetService<IArcServices>();
+            var result = await arcService.DisconnectFromAzureArcAsync(
+                options.ResourceGroupName!,
+                options.ClusterName!,
+                options.UserProvidedPath!);
```

Comment:
User Provided Path is a required param as per description, should this be validated here ?
Its specified as nullable here.